xapian-core/api/queryinternal.cc

```diff
@@ -1314,7 +1379,7 @@ QueryValueGE::postlist(QueryOptimiser *qopt, double factor) const
     if (limit > db.get_value_upper_bound(slot)) {
 	RETURN(NULL);
     }
-    if (limit < lb) {
+    if (limit <= lb) {    // should't this be <= ?? 
```

Comment:
This change is unrelated to your other changes. If this change should be made, it should be done at least as a separate commit, with a testcase that fails before the change happens. (It could also be done as entirely separate PR.)

Reply:
(If the change is about efficiency rather than correctness, then a test case is going to be hard to impossible to write. This change should still be in its own commit in that case.)